src/js/heuristicblocking.js

```diff
@@ -125,12 +127,115 @@ HeuristicBlocker.prototype = {
       return {};
     }
 
-    // ignore if there are no tracking cookies
-    if (!hasCookieTracking(details, origin)) {
+    // check if there are tracking cookies
+    if (hasCookieTracking(details, origin)) {
+      this._recordPrevalence(fqdn, origin, tabOrigin);
       return {};
     }
 
-    this._recordPrevalence(fqdn, origin, tabOrigin);
+    let self = this;
+
+    // check if this request is a cookie-syncing pixel
+    chrome.cookies.getAll({}, function(cookies) {
+      self.pixelCookieSyncAccounting(details, cookies);
+    });
+  },
+
+  /**
+   * Checks for cookie syncing: requests to third-party domains that include
+   * high entropy data from first-party cookies.
+   *
+   * TODO: is it OK to call this from heuristicBlockingAccounting?
+   *
+   * @param details are those from onBeforeSendHeaders
+   * @param cookies are the result of chrome.cookies.getAll()
+   * @returns {*}
+   */
+  pixelCookieSyncAccounting: function (details, cookies) {
+    // only interested in images
+    if (details.type != 'image') {
+      return false;
+    }
+
+    let fqdn = (new URI(details.url)).host,
+      origin = window.getBaseDomain(fqdn),
+      tabOrigin = tabOrigins[details.tabId],
+      initiator = tabURLs[details.tabId],
+      args = _extractArgs(details),
+      TRACKER_ENTROPY_THRESHOLD = 33,
+      MIN_STR_LEN = 8;
+
+    for (let key in args) {
+      if (!args.hasOwnProperty(key)) {
+        continue;
+      }
+
+      let value = args[key];
+
+      // the argument must be sufficiently long
+      if (!value || value.length < constants.MIN_ARG_LEN) {
+        continue;
+      }
+
+      let substrings, maxString, maxEntropy;
+      // check if this argument is derived from a high-entropy first-party cookie
+      for (let cookie of cookies) {
+        if (cookie.domain == origin) {
+          continue;
+        }
+
+        // find the longest common substring between this arg and the cookies
+        // associated with the document
+        substrings = utils.findCommonSubstrings(cookie.value, value) || [];
+        maxEntropy = 0;
+        for (let s of substrings) {
+          // ignore the substring if it's part of the first-party URL
+          if (initiator.indexOf(s) != -1) {
+            continue;
+          }
+
+          // or the user agent string
```

Comment:
Document with an example please. Not immediately obvious why that's not interesting.